presto-main-base/src/main/java/com/facebook/presto/sql/planner/PlannerUtils.java

```diff
@@ -480,7 +482,16 @@ public static boolean isSupportedArrayContainsFilter(FunctionResolution function
             if (functionResolution.isArrayContainsFunction(callExpression.getFunctionHandle())) {
                 RowExpression array = callExpression.getArguments().get(0);
                 RowExpression element = callExpression.getArguments().get(1);
-                checkState(array.getType() instanceof ArrayType && ((ArrayType) array.getType()).getElementType().equals(element.getType()));
+                // varchar(n) is not supported in Presto C++, compare base types of array and element for varchar to
```

Comment:
Can you move this commit outside of this PR and write a separate individual test for it.  You can start with the original test you fixed, and then we might want to add more variations to it.